scipy/optimize/_hessian_update_strategy.py

```diff
@@ -253,10 +259,12 @@ class BFGS(FullHessianUpdateStrategy):
         unaffected by the exception strategy. By default is equal to
         1e-8 when ``exception_strategy = 'skip_update'`` and equal
         to 0.2 when ``exception_strategy = 'damp_update'``.
-    init_scale : {float, 'auto'}
+    init_scale : {float, np.array, 'auto'}
         Matrix scale at first iteration. At the first
         iteration the Hessian matrix or its inverse will be initialized
         with ``init_scale*np.eye(n)``, where ``n`` is the problem dimension.
+        The scale can be any array-like object that can be multiplied
+        with `np.eye(n)`.
```

Comment:
This means I can have vectors that can be broadcasted or elementwise-multiplied square matrices? We need some explanation here like anything that can be converted to a float scalar or an nxn matrix or whatever the size can be. The edge cases are more important than the happy-flow to prevent issues later

Reply:
Yes exactly, that's the idea, but I'll expand the text